assets/app/features/upgrade/steps.factory.js

```diff
@@ -95,5 +98,66 @@
                 }
             ];
         }
+
+        function stepByState(state) {
+            return _.find(factory.steps, function (step) { return step.state === state; });
+        }
+
+        function stepByID(id) {
+            return _.find(factory.steps, function (step) { return step.id === id; });
+        }
+
+        function lastStateForRestore(statusData) {
+            var stepToStateMap = {
+                'upgrade_prechecks': 'upgrade-landing',
+                'upgrade_prepare': 'upgrade-landing',
+                'admin_backup': 'upgrade.backup',
+                'admin_repo_checks': 'upgrade.administration-repository-checks',
+                'admin_upgrade': 'upgrade.upgrade-administration-server',
+                'database': 'upgrade.database-configuration',
+                'nodes_repo_checks': 'upgrade.nodes-repositories-checks',
+                'nodes_services': 'upgrade.openstack-services',
+                'nodes_db_dump': 'upgrade.openstack-services',
+                'nodes_upgrade': 'upgrade.upgrade-nodes',
+                'finished': 'TODO(skazi): no UI for this step yet',
```

Comment:
Once the nodes upgrade is completed, the user will see a "Finish" button in the navbar (instead of "Next"). On click, will be redirected (still under discussion) to crowbar dashboard or barclamps page if couldn't be reapplied automatically.

So I guess the state is `upgrade.upgrade-nodes` for this case.